src/components/prototypes/ContractorAddressForm/ContractorAddressForm.tsx

```diff
@@ -0,0 +1,149 @@
+import type { ReactNode } from 'react'
+import { FormProvider, useForm } from 'react-hook-form'
+import { zodResolver } from '@hookform/resolvers/zod'
+import { useTranslation } from 'react-i18next'
+import { ContractorAddressFormProvider } from './ContractorAddressFormProvider'
+import { useContractorAddressForm, ContractorAddressFormSchema } from './useContractorAddressForm'
+import type { ContractorAddressFormValues } from './useContractorAddressForm'
+import { Form as HtmlForm } from '@/components/Common/Form/Form'
+import { useI18n, useComponentDictionary } from '@/i18n'
+import { Flex, TextInputField, SelectField, Grid } from '@/components/Common'
+import { ActionsLayout } from '@/components/Common/ActionsLayout/ActionsLayout'
+import type { BaseComponentInterface } from '@/components/Base'
+import { BaseUIComponent } from '@/components/Base/Base'
+import { useComponentContext } from '@/contexts/ComponentAdapter/useComponentContext'
+import { contractorEvents, STATES_ABBR } from '@/shared/constants'
+import { useBase } from '@/components/Base/useBase'
+
+export interface ContractorAddressProps extends BaseComponentInterface<'Contractor.Address'> {
+  contractorId: string
+  defaultValues?: ContractorAddressFormValues
+  children?: ReactNode
+  className?: string
+}
+
+function ContractorAddress({ defaultValues, ...props }: ContractorAddressProps) {
+  return (
+    <ContractorAddressFormProvider {...props}>
+      <BaseUIComponent>
+        <Root defaultValues={defaultValues} {...props} />
+      </BaseUIComponent>
+    </ContractorAddressFormProvider>
+  )
+}
+
+function Root({ defaultValues, children, className, dictionary }: ContractorAddressProps) {
+  useComponentDictionary('Contractor.Address', dictionary)
+  useI18n('Contractor.Address')
+
+  const {
+    contractor,
+    defaultValues: formDefaultValues,
+    isUpdating,
+    onSubmit,
+  } = useContractorAddressForm()
+  const contractorType = contractor?.type
+
+  const Components = useComponentContext()
+  const { onEvent } = useBase()
+
+  const { t } = useTranslation('Contractor.Address')
+
+  const composedDefaultValues = {
+    street1: formDefaultValues.street1 || defaultValues?.street1,
+    street2: formDefaultValues.street2 || defaultValues?.street2,
+    city: formDefaultValues.city || defaultValues?.city,
+    state: formDefaultValues.state || defaultValues?.state,
+    zip: formDefaultValues.zip || defaultValues?.zip,
+  }
```

Comment:
Not sure if it's ultimately worth it to have the hook supply the default values and merge them here via props. Only thought was, sometimes composing the default values gets complicated